src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SmkReferenceExpression.kt

```diff
@@ -29,15 +27,22 @@ class SmkReferenceExpression(node: ASTNode): PyElementImpl(node), PsiNamedElemen
     private fun getNameNode() = getIdentifierNode(node)
 
     private class SmkRuleOrCheckpointNameReference(
-            element: PsiElement,
+            private val element: PsiNamedElement,
             textRange: TextRange
-    ) : PsiReferenceBase<PsiElement>(element, textRange) {
+    ) : PsiReferenceBase<PsiElement>(element, textRange), PsiPolyVariantReference {
         private val key: String = element.text
 
         override fun resolve(): PsiElement? =
```

Comment:
Better to implement resolve using multi resolve, e.g. `override fun resolve() = multiResolve(false).firstOrNull()?.element`